app/controllers/extractors_controller.rb

```diff
@@ -32,10 +32,13 @@ def create
     extractor_class = Extractor.of_type(params[:extractor][:type])
     @extractor = extractor_class.new(extractor_params(extractor_class))
 
-    @extractor.save
     respond_to do |format|
+      if @extractor.save
+        format.json { render json: @extractor }
+      else
+        format.json { render json: @extractor.errors, status: :unprocessable_entity }
+      end
       format.html { respond_with @extractor, location: workflow_path(workflow, anchor: 'extractors') }
-      format.json { respond_with @extractor }
```

Comment:
realized that if we just did `format.json { render json: @extractor }` on error, would still return 200 status.